app/Console/Kernel.php

```diff
@@ -30,62 +30,70 @@ protected function schedule(Schedule $schedule): void
+            $schedule->job(new CleanupInstanceStuffsJob)->everyMinute()->onOneServer()->timezone($serverTimezone);
 
             // Server Jobs
-            $this->check_scheduled_backups($schedule);
-            $this->check_resources($schedule);
-            $this->check_scheduled_tasks($schedule);
-            $schedule->command('uploads:clear')->everyTwoMinutes();
+            $this->check_scheduled_backups($schedule, $serverTimezone);
+            $this->check_resources($schedule, $serverTimezone);
+            $this->check_scheduled_tasks($schedule, $serverTimezone);
+            $schedule->command('uploads:clear')->everyTwoMinutes()->timezone($serverTimezone);
         } else {
             // Instance Jobs
-            $schedule->command('horizon:snapshot')->everyFiveMinutes();
-            $schedule->command('cleanup:unreachable-servers')->daily();
-            $schedule->job(new PullCoolifyImageJob)->cron($settings->update_check_frequency)->onOneServer();
-            $schedule->job(new PullTemplatesFromCDN)->cron($settings->update_check_frequency)->onOneServer();
-            $schedule->job(new CleanupInstanceStuffsJob)->everyTwoMinutes()->onOneServer();
-            $this->schedule_updates($schedule);
+            $schedule->command('horizon:snapshot')->everyFiveMinutes()->timezone($serverTimezone);
+            $schedule->command('cleanup:unreachable-servers')->daily()->timezone($serverTimezone);
+            $schedule->job(new PullCoolifyImageJob)->cron($settings->update_check_frequency)->onOneServer()->timezone($serverTimezone);
+            $schedule->job(new PullTemplatesFromCDN)->cron($settings->update_check_frequency)->onOneServer()->timezone($serverTimezone);
+            $schedule->job(new CleanupInstanceStuffsJob)->everyTwoMinutes()->onOneServer()->timezone($serverTimezone);
+            $this->schedule_updates($schedule, $serverTimezone);
 
             // Server Jobs
-            $this->check_scheduled_backups($schedule);
-            $this->check_resources($schedule);
-            $this->pull_images($schedule);
-            $this->check_scheduled_tasks($schedule);
+            $this->check_scheduled_backups($schedule, $serverTimezone);
+            $this->check_resources($schedule, $serverTimezone);
+            $this->pull_images($schedule, $serverTimezone);
+            $this->check_scheduled_tasks($schedule, $serverTimezone);
 
-            $schedule->command('cleanup:database --yes')->daily();
-            $schedule->command('uploads:clear')->everyTwoMinutes();
+            $schedule->command('cleanup:database --yes')->daily()->timezone($serverTimezone);
+            $schedule->command('uploads:clear')->everyTwoMinutes()->timezone($serverTimezone);
         }
     }
 
-    private function pull_images($schedule)
+    private function getServerTimezone()
+    {
+        $server = Server::find(0); // Only main server is used for scheduling tasks, not each server timezone?
```

Comment:
Should we get each server timezone for scheduled task? How would we determine the destination server?